standalone-metastore/metastore-server/src/main/java/org/apache/hadoop/hive/metastore/HiveMetaStore.java

```diff
@@ -6335,22 +6344,42 @@ private int get_num_partitions_by_expr(final String catName, final String dbName
 
     @Override
     public List<Partition> get_partitions_by_names(final String dbName, final String tblName,
-                                                   final List<String> partNames) throws TException {
+           final List<String> partNames, boolean getColStats) throws TException {
 
-      String[] parsedDbName = parseDbName(dbName, conf);
+      String[] dbNameParts = parseDbName(dbName, conf);
+      String parsedCatName = dbNameParts[CAT_NAME];
+      String parsedDbName = dbNameParts[DB_NAME];
       List<Partition> ret = null;
       Exception ex = null;
-
-      startTableFunction("get_partitions_by_names", parsedDbName[CAT_NAME], parsedDbName[DB_NAME],
-          tblName);
+      startTableFunction("get_partitions_by_names", parsedCatName, parsedDbName,
+              tblName);
       try {
-        authorizeTableForPartitionMetadata(parsedDbName[CAT_NAME], parsedDbName[DB_NAME], tblName);
+        authorizeTableForPartitionMetadata(parsedCatName, parsedDbName, tblName);
 
-        fireReadTablePreEvent(parsedDbName[CAT_NAME], parsedDbName[DB_NAME], tblName);
+        fireReadTablePreEvent(parsedCatName, parsedDbName, tblName);
 
-        ret = getMS().getPartitionsByNames(parsedDbName[CAT_NAME], parsedDbName[DB_NAME], tblName,
-            partNames);
+        ret = getMS().getPartitionsByNames(parsedCatName, parsedDbName, tblName, partNames);
```

Comment:
I agree with that. The name "ret" already existed, I changed a different part of that function call, since it eased the new code that was added. Again for the sake of easy back-port, I would leave that change aside.

Reply:
I noticed, you use ret in the new code. So, suggested to rename it as it makes the code readable. Anyways, your call.